gce/survival-training/gce/startup.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+GCE_USER=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/gceUser -H "Metadata-Flavor: Google")
+TRAINER_REPO=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/trainerRepo -H "Metadata-Flavor: Google")
+TRAINER_MODULE=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/trainerModule -H "Metadata-Flavor: Google")
+
+JOB_DIR=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/jobDir -H "Metadata-Flavor: Google")
+
+TRAIN_STEPS=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/trainSteps -H "Metadata-Flavor: Google")
+CHECKPOINT_STEPS=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/checkpointSteps -H "Metadata-Flavor: Google")
+HPARAM1=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/hparam1 -H "Metadata-Flavor: Google")
+HPARAM2=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/hparam2 -H "Metadata-Flavor: Google")
+
+KEEP_ALIVE=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/keepAlive -H "Metadata-Flavor: Google")
+
+cd "/home/$GCE_USER"
+
+gcloud source repos clone "$TRAINER_REPO"
+
+cd "/home/$GCE_USER/$TRAINER_REPO"
+
+git pull origin master
+
+mkdir -p $JOB_DIR
+
+sudo -u $GCE_USER python -m $TRAINER_MODULE \
```

Comment:
Why do you need sudo? And below as well.

Reply:
To run as $GCE_USER instead of root.

Reply:
I understand that, but why is sudo needed. Why won't the code run as a GCE_USER?

Reply:
Because the startup script runs as root.

Either you have to modify your environment as root before you froze the VM image and therefore avoid this game in the startup script or you have to modify the user environment pre-freeze and access it in this manner in the startup script. I went for the latter because it feels safer.

Reply:
Why does the startup script run as root?

Reply:
That's how GCE is set up.